client/sites/components/sites-dataviews/index.tsx

```diff
@@ -50,32 +48,52 @@ const DotcomSitesDataViews = ( {
 	paginationInfo,
 	dataViewsState,
 	setDataViewsState,
+	selectedItem,
+	openSitePreviewPane,
 }: Props ) => {
 	const { __ } = useI18n();
 	const userId = useSelector( getCurrentUserId );
 
-	const openSitePreviewPane = useCallback(
-		( site: SiteExcerptData ) => {
-			setDataViewsState( ( prevState: DataViewsState ) => ( {
-				...prevState,
-				selectedItem: site,
-				type: 'list',
-			} ) );
-		},
-		[ setDataViewsState ]
-	);
+	// Scroll to selected site in the list when in list view.
+	const scrollContainerRef = useRef< HTMLElement >();
+	const previousDataViewsState = usePrevious( dataViewsState );
+	const previousSelectedItem = usePrevious( selectedItem );
+	useLayoutEffect( () => {
+		if ( ! scrollContainerRef.current || previousDataViewsState?.type !== dataViewsState.type ) {
+			scrollContainerRef.current = document.querySelector( '.dataviews-view-list' ) as HTMLElement;
+		}
+
+		if ( ! previousSelectedItem && selectedItem && dataViewsState.type === 'list' ) {
+			window.setTimeout(
+				() => scrollContainerRef.current?.querySelector( 'li.is-selected' )?.scrollIntoView(),
+				300
+			);
+			return;
+		}
+
+		if ( previousDataViewsState?.page !== dataViewsState.page ) {
+			scrollContainerRef.current?.scrollTo( 0, 0 );
+		}
+	}, [
+		dataViewsState.type,
+		dataViewsState.page,
+		selectedItem,
+		previousDataViewsState,
+		previousSelectedItem,
+	] );
```

Comment:
Pulled in from [ItemsDataViews](https://github.com/Automattic/wp-calypso/blob/update/dv/client/a8c-for-agencies/components/items-dashboard/items-dataviews/index.tsx#L61)